### ISSUE TITLE
net/url: invalid control character in url

### ISSUE THREAD

**Issue by rew1nter** [go-resty/resty#533]
```
2022/03/31 12:20:36.092662 ERROR RESTY parse "https://example.com/j/\v94684797417": net/url: invalid control character in URL, Attempt 1
[WARN] upload request failed for https://examle.com/j/
                                                                         94684797417 : parse "https://example.com/j/\v94684797417": net/url: invalid control character in URL
2022/03/31 12:20:36.092776 ERROR RESTY parse "https://example.com/j/\v94684797417/": net/url: invalid control character in URL, Attempt 1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x660b30]

goroutine 114288 [running]:
github.com/go-resty/resty.(*Response).StatusCode(...)
        /root/audacity/go/src/github.com/go-resty/resty/response.go:52
main.makerequest({0xc02dd28880, 0xc02b7947b0}, {0x0, 0x0}, 0x0)
```

I tried to catch this using client.onError but failed. I wanna continue execution after catching this. How do you catch this error?

**Comment by segevda** [comment]
I hope I understand your question, it's hard to follow without a code snippet.

Not sure which resty method you're using so I'm assuming Get(), which returns (resp *Response, err error). Since the request failed, resp is nil, so calling StatusCode() on it panics. However - err in this case shouldn't be nil, and checking it will help you "catch" the error.

I hope this answers your question.

**Comment by Jeongseup** [comment]
I used recover func in go to handle this error and avoid this problem.

1. first of all, you should add some recovery function.
example like
```go
// in your code
func handleOutOfNilResponse() {
	if r := recover(); r != nil {
		fmt.Errorln("Recovering from panic:", r)
		fmt.Infoln("the recover handles the go-resty nil response' panics gracefully")
	}
}
```
2. and you should the function with defer keyword 
```go
defer handleOutOfNilResponse()

resp, err := requester.EnableTrace().Get()
if err != nil {
	if resp == nil {
		fmt.Errorln("passed resp.Time() nil point err")
		// avoid go-resty panic
		return
		}
fmt.Errorln("  Time       :", resp.Time())
fmt.Errorln("  Error      :", err)
return
}

```

**Comment by segevda** [comment]
@Jeongseup Hi, I didn't raise the issue, @rew1nter did.
But since you mentioned me, I'll just add that using `recover` is wrong in this case (and probably in most cases), and might obscure other errors and create complex spaghetti code.

If you check the error (and even do a nil check on resp, which is also not required here), there is no reason to use recover. If this program still panics, it means there's a bug here and hiding it with `recover` is just, well, wrong.

**Comment by jeevatkm** [comment]
@rew1nter @Jeongseup Typically, if the error is non-nil, then it means the response is nil, so there is no need to access the response object. 
Also, could you go through @segevda's inputs? He explained why you should not use recover.
